builtin/providers/azurerm/resource_arm_cdn_profile.go

```diff
@@ -0,0 +1,186 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/cdn"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmCdnProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmCdnProfileCreate,
+		Read:   resourceArmCdnProfileRead,
+		Update: resourceArmCdnProfileUpdate,
+		Delete: resourceArmCdnProfileDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"sku": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCdnProfileSku,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmCdnProfileCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	cdnProfilesClient := client.cdnProfilesClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM CDN Profile creation.")
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	sku := d.Get("sku").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := cdn.ProfilePropertiesCreateParameters{
+		Sku: &cdn.Sku{
+			Name: cdn.SkuName(sku),
+		},
+	}
+
+	cdnProfile := cdn.ProfileCreateParameters{
+		Location:   &location,
+		Properties: &properties,
+		Tags:       expandTags(tags),
+	}
+
+	resp, err := cdnProfilesClient.Create(name, cdnProfile, resGroup)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*resp.ID)
+
+	log.Printf("[DEBUG] Waiting for CDN Profile (%s) to become available", name)
+	stateConf := &resource.StateChangeConf{
```

Comment:
Once the Storage Account work gets merged in I wonder if we can use the polling request here - perhaps not though as it seems to be a state that will update and this might be the more appropriate way.